### PR TITLE
Fix cache error loosing response body.

### PR DESCRIPTION
**Pull request by Torgan**

### CODE REVIEW

**Comment by csarrazi** [comment]
Hum, I think I know what was wrong with the response body. Could you try adding a $response->getBody()->seek(0) instead ?

What happens is that when the response has been read once, the stream's cursor is at the end of the content, which means that fetching the content once again will return an empty string.

**Comment by Torgan** [comment]
Yes that was also my conclusion. I'll try the ->seek(0) which looks better.

**Comment by Torgan** [comment]
Here we go

**Comment by csarrazi** [comment]
Thank you @Torgan ! :)

**Comment by Torgan** [comment]
My pleasure :)

File: src/GuzzleHttp/Middleware.php [hunk 1]
```diff
@@ -55,6 +55,8 @@ public static function cache(StorageAdapterInterface $adapter, $debug = false)
                         function (ResponseInterface $response) use ($adapter, $request, $debug) {
                             $adapter->save($request, $response);
 
+                            $response->getBody()->seek(0);
```
Comment (SofHad) [inline src/GuzzleHttp/Middleware.php]: We can use this function  https://github.com/guzzle/psr7/blob/master/src/functions.php#L250
> The body of the message will only be rewound if a call to `tell()` returns a value other than `0`.

**Comment by lunetics** [comment]
Maybe i'm doing something wrong. Just curious. I use a dedicated client with the bazinga geocoder bundle. Also using the guzzle6 http adapter with the dedicated client. Geocoding works, but in the guzzle bundle debug tab i cannot see the response body. also the datacollector is called in an extra request and the psr7 Stream is detached (no content)

any idea(s)?

**Comment by csarrazi** [comment]
Could you create a separate issue, and elaborate a little bit more?
Do you use the `cache` subscriber?
Which version of the bundle are you using?

**Pull request merged**